app/assets/javascripts/common/reports.js

```diff
@@ -570,12 +570,16 @@ function stackedBarChartTooltip(config) {
     visitButNoBPMeasureRateNode.innerHTML = dataPoints[2].value + "%";
     uncontrolledRateNode.innerHTML = dataPoints[1].value + "%";
     controlledRateNode.innerHTML = dataPoints[0].value + "%";
-    missedVisitsPatientsNode.innerHTML = missedVisitsPatients[dataPoints[0].index];
-    visitButNoBPMeasurePatientsNode.innerHTML = visitButNoBPMeasurePatients[dataPoints[0].index];
-    uncontrolledPatientsNode.innerHTML = uncontrolledPatients[dataPoints[0].index];
-    controlledPatientsNode.innerHTML = controlledPatients[dataPoints[0].index];
+    missedVisitsPatientsNode.innerHTML = formatNumberWithCommas(missedVisitsPatients[dataPoints[0].index]);
+    visitButNoBPMeasurePatientsNode.innerHTML = formatNumberWithCommas(visitButNoBPMeasurePatients[dataPoints[0].index]);
+    uncontrolledPatientsNode.innerHTML = formatNumberWithCommas(uncontrolledPatients[dataPoints[0].index]);
+    controlledPatientsNode.innerHTML = formatNumberWithCommas(controlledPatients[dataPoints[0].index]);
     periodStartNodes.forEach(node => node.innerHTML = periodInfo[dataPoints[0].index].bp_control_start_date);
     periodEndNodes.forEach(node => node.innerHTML = periodInfo[dataPoints[0].index].bp_control_end_date);
-    cumulativeRegistrationsNodes.forEach(node => node.innerHTML = adjustedRegistrations[dataPoints[0].index]);
+    cumulativeRegistrationsNodes.forEach(node => node.innerHTML = formatNumberWithCommas(adjustedRegistrations[dataPoints[0].index]));
   }
+}
+
+function formatNumberWithCommas(value) {
+  return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```

Comment:
@claudiovallejo Could you post a screenshot of how the numbers must be formatted to see if we can find a simpler solution to this? It's hard to parse this regex to figure out what it's trying to do.

Reply:
Also, I'm assuming this will eventually have to be localized, hence, if possible, we should try to do this in Ruby.

Reply:
@kitallis I would add a screenshot, but fake data in SBX are all less than 1,000 so no commas are displayed. But here's an example: `12345 -> 12,345`.

Is it ok if we fix in JS now? I'm doing some cleanup work today and tomorrow and can ask Rob's help to move this to Ruby.